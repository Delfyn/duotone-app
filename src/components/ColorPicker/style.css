.color-field {
  width: 88px;
  height: 40px;
  border-radius: 2rem;
  font-size: .75rem;
  color: #fff;
  text-align: center;
  line-height: 40px;
  cursor: pointer;
}
