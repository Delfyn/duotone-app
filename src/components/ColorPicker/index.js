/* @flow */

import React, { Component } from 'react';
import { ChromePicker } from 'react-color';
import classNames from 'classnames';
import './style.css';

class ColorPicker extends Component {
  constructor() {
    super();
    this.state = {
      displayColorPicker: false,
      color: '#fcf0ff',
    };

    this.handleClick = () => {
      this.setState({
        displayColorPicker: !this.state.displayColorPicker,
      });
    };
    this.handleClose = () => {
      this.setState({
        displayColorPicker: false,
      });
    };
  }
  state = {
    displayColorPicker: Boolean,
    color: String,
  };
  handleClick: () => void
  handleClose: () => void
  render() {
    const popover = {
      position: 'absolute',
      zIndex: '5',
    };
    const cover = {
      position: 'fixed',
      top: '-1000px',
      right: '-1000px',
      bottom: '-1000px',
      left: '-1000px',
    };
    return (
      <div>
        <div
          className={classNames('color-field flex-auto', this.props.className)}
          onClick={this.handleClick}
          style={{ backgroundColor: this.props.color }}
        >
          {this.props.color}
        </div>
        { this.state.displayColorPicker ? <div style={popover}>
          <div style={cover} onClick={this.handleClose} />
          <ChromePicker
            color={this.props.color}
            onChangeComplete={color => this.props.onChange(color.hex)}
            disableAlpha
          />
        </div> : null }
      </div>
    );
  }
}

export default ColorPicker;
