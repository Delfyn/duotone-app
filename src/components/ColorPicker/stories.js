/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import ColorPicker from './index';

storiesOf('ColorPicker', module).add('Simple example', () => (
  <ColorPicker color="#fd3659" onChange={action('onChange')} />
));
