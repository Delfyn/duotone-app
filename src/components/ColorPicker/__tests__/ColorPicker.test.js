import React from 'react';
import renderer from 'react-test-renderer';
import ColorPicker from '../index';

test('ColorPicker renders correctly', () => {
  const component = renderer.create(<ColorPicker />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
