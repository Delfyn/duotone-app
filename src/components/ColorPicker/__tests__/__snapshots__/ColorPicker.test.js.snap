exports[`test ColorPicker renders correctly 1`] = `
<div>
  <div
    className="color-field flex-auto"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    } />
</div>
`;
